cmd/clusterctl/client/cluster/mover_test.go

```diff
@@ -248,6 +248,59 @@ var moveTests = []struct {
 		},
 		wantErr: false,
 	},
+	{
+		name: "Cluster with MachineDeployment with a static bootstrap config",
+		fields: moveTestsFields{
+			objs: test.NewFakeCluster("ns1", "cluster1").
+				WithMachineDeployments(
+					test.NewFakeMachineDeployment("md1").
+						WithStaticBootstrapConfig().
+						WithMachineSets(
+							test.NewFakeMachineSet("ms1").
+								WithMachines(
+									test.NewFakeMachine("m1"),
+									test.NewFakeMachine("m2"),
+								),
+						),
+				).Objs(),
+		},
+		wantMoveGroups: [][]string{
+			{ // group 1
+				"cluster.x-k8s.io/v1beta1, Kind=Cluster, ns1/cluster1",
+			},
+			{ // group 2 (objects with ownerReferences in group 1)
+				// owned by Clusters
+				"/v1, Kind=Secret, ns1/cluster1-ca",
+				"/v1, Kind=Secret, ns1/cluster1-kubeconfig",
+				"cluster.x-k8s.io/v1beta1, Kind=MachineDeployment, ns1/md1",
+				"bootstrap.cluster.x-k8s.io/v1beta1, Kind=GenericBootstrapConfigTemplate, ns1/md1",
```

Comment:
If we are using a statis bootstrap then there should not be any `GenericBootstrapConfigTemplate` in the output. Isn't it?
(Same applies for the 2 `GenericBootstrapConfig` entries below.)

Looks like this is because even if we are using a static bootstrap config we are creating and adding the BootstrapConfigTemplate and the associated BootstrapConfigs into the output object [here](https://github.com/ykakarap/cluster-api/blob/ba83907ee1749ea38652fca55cb99f5fa2014cc8/cmd/clusterctl/internal/test/fake_objects.go#L667). This should ideally be adjusted to reflect a case where we only have the associated secrets and not the BootstrapConfigs.

Reply:
If we are using a MachineDeployment/MachineSet with a static bootstrap (.bootstrap.configRef == nil) then no `KubedmConfig` would be created when the MachineSet creates the required Machines.

I am curious in the case you shared here was the KubeadmConfig created by the MachineDeployment/MachineSet while the MachineDeployment/MachineSet only has `dataSecretName` under `bootstrap` and no `configRef`? 🤔

Reply:
i'm very sorry for the confusion :see_no_evil: but i've turned wrong when i analyzed the `machineset|machinedeployment` where `.bootstrap.configRef == nil`.

the machine `anzu-2qpjh` is a `KCP` based machine where the config is different.

if i check the correct machine, the config is exactly as it's expected:

```yaml
spec:
  bootstrap:
    dataSecretName: anzu-md-ignition
  clusterName: anzu
  infrastructureRef:
    [...]
```

Reply:
> If we are using a MachineDeployment/MachineSet with a static bootstrap (.bootstrap.configRef == nil) then no `KubedmConfig` would be created when the MachineSet creates the required Machines.

@ykakarap i've changed the fake machine Obj implementation to make this case work. PTAL :pray: